pulsar-client-go/pulsar/client.go

```diff
@@ -103,16 +103,22 @@ type Client interface {
 	// This method will block until the producer is created successfully
 	CreateProducer(ProducerOptions) (Producer, error)
 
+	CreateTypedProducer(ProducerOptions, Schema) (Producer, error)
```

Comment:
Could we follow same convention across the 3 methods, using `WithSchema()` ?

Reply:
sure, good suggestion